cores/esp8266/core_esp8266_i2s.c

```diff
@@ -92,10 +93,15 @@ void ICACHE_FLASH_ATTR i2s_slc_isr(void) {
       i2s_slc_queue_next_item(); //free space for finished_item
     }
     i2s_slc_queue[i2s_slc_queue_len++] = finished_item->buf_ptr;
+    if (i2s_callback) i2s_callback();
     ETS_SLC_INTR_ENABLE();
   }
 }
 
+void ICACHE_FLASH_ATTR i2s_set_callback(void (*callback) (void)){
```

Comment:
Now that 2.4.1 is out this can get into the merge queue.  Only minor change I would request is to remove the ICACHE_FLASH_ATTR from the set-callback function.  There's no need for this to be kept in precious IRAM.